pdf_rendering.py

```diff
@@ -1,239 +1,378 @@
-
-def create_total(pdf,
-                invoice,
-                tableColumnDateWidth, 
-                tableColumnClientCodeWidth,
-                tableColumnDescriptionWidth,
-                tableColumnHoursWidth,
-                ):
-    pdf.set_y(pdf.get_y()+5)
-    pdf.set_x(tableColumnDateWidth + tableColumnClientCodeWidth + tableColumnDescriptionWidth+10)
-    pdf.set_font(family='Arial',style='B')
-    pdf.cell(w=20, txt=f'Total',align='C')
-
-    totalhours = 0
-    for task in invoice.activity.tasks:
-        totalhours += int(task.duration)
+    def render_total():
+        pdf.set_y(pdf.get_y() + 5)
+        pdf.set_x(table_column_date_width + table_column_client_code_width + table_column_description_width+10)
+        pdf.set_font(family='Arial', style='B')
+        pdf.cell(w=20, txt=f'Total', align='C')
+
+        pdf.set_x(pdf.get_x() + 5)
+        pdf.cell(w=table_column_hours_width, txt=f'{invoice.activity.duration}')
         
-    pdf.set_x(pdf.get_x() + 5)
-    pdf.cell(w=tableColumnHoursWidth, txt=f'{totalhours}')
+        pdf.set_font(family='Arial')
+    
+    def render_seller_signature():
+        current_y = pdf.get_y()
+        pdf.set_xy(initialX + 30, pdf.get_y() + 30)
+        pdf.cell(w=20, txt=f'FURNIZOR: ', align = 'C')
+        pdf.set_y(pdf.get_y() + 7)
+        pdf.set_x(initialX + 30)
+        pdf.cell(w=20, h=6, txt=f'{invoice.seller.name}',align = 'C', ln = 1)
+        pdf.set_x(initialX + 30)
+        pdf.cell(w=20, h=6, txt=f'Florinel',align = 'C', ln = 1)
+        pdf.set_x(initialX + 30)
+        pdf.cell(w=20, h=6, txt=f'L.S.', align = 'C', ln = 1)
+
+        pdf.set_y(current_y)
+
+    def render_buyer_signature():
+        pdf.set_xy(initialX + 130, pdf.get_y() + 30)
+        pdf.cell(w=20, txt=f'BENEFICIAR: ', align = 'C')
+        pdf.set_y(pdf.get_y() + 7)
+        pdf.set_x(initialX + 130)
+        pdf.cell(w=20, h=6, txt=f'{invoice.buyer.name}',align = 'C', ln = 1)
+        pdf.set_x(initialX + 130)
+        pdf.cell(w=20, h=6, txt=f'Mariusel', align = 'C', ln = 1)
+        pdf.set_x(initialX + 130)
+        pdf.cell(w=20, h=6, txt=f'L.S.', align = 'C', ln = 1)
+    
+    def process_text_to_fit_width(pdf, text,width):
+        words = str(text).split()
+        rows = []
+        row = ''
+
+        for word in words:
+            next_row = ' '.join([row, word])
+            if pdf.get_string_width(next_row) > width:
+                # start a new row
+                rows.append(row)
+                row = next_row
+        rows.append(row)
+
+        return rows
+
+    render_seller_info()
+    render_buyer_info()
+    render_title()
+    render_table_headings()
+    render_tasks()
+    render_total()
+    render_seller_signature()
+    render_buyer_signature()
+    write_to_page(pdf, document_name)
+
+
+def render_invoice(invoice):
+    pdf = fpdf.FPDF(orientation='P', unit='mm', format='A4')
+    pdf.add_page()
+    pdf.set_font('Arial')
+
+    document_name = 'demo_pdf_invoice.pdf'
```

Comment:
We should find a way to specify the generated documents name (both report and invoice).